### PR TITLE
Refactor to resource partials

### PR DESCRIPTION
**Pull request by bmhughes**
### Description

- Refactor to use resource partials
- General refactor to improve maintainability

### Issues Resolved

- n/a

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: resources/install.rb [hunk 1]
```diff
@@ -1,39 +1,74 @@
...
+
+property :use_lua, [true, false],
+          default: false
+
+property :lua_lib, String
+
+property :lua_inc, String
+
+property :use_systemd, [true, false],
+          default: lazy { source_version.to_f >= 1.8 }
```
Comment (damacus) [inline resources/install.rb:71]: This lazy can go away now probably

Comment (bmhughes) [reply resources/install.rb:71]: The tables turn 😂. Although i'll just double check but I'm 99% other properties are another edge case where that needs to stay.

Comment (bmhughes) [reply resources/install.rb:71]: Yes unfortunately

```
NameError
---------
undefined local variable or method `source_version' for #<Class:0x00000000036b8b60>
```

**Review by damacus — APPROVED** [review]

**Comment by damacus** [comment]
md-links are failing because the target content doesn't yet exist.

File: metadata.rb [hunk 2]
```diff
@@ -3,10 +3,10 @@
 maintainer_email  'help@sous-chefs.org'
 license           'Apache-2.0'
 description       'Installs and configures haproxy'
-version           '11.0.0'
+version           '12.0.0'
```
[the remark this answers is not in the document]
Comment (bmhughes) [reply metadata.rb:6]: Yeh my bad I've pushed it to my internal supermarket and commited that by accident. It'll be gone shortly.

**Review by damacus — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [12.0.0](https://github.com/sous-chefs/haproxy/releases/tag/12.0.0)

**Pull request merged**